Author: Maxim Paperno <MPaperno@WorldDesign.com>
Description: Set homedir to Debian standard.

Index: spampd/spampd.pl
===================================================================
--- spampd.orig/spampd.pl
+++ spampd/spampd.pl
@@ -461,8 +461,9 @@ sub new {
     assassin => {
       debug                => 0,                  # debug flag, can be boolean or a list to pass to SA (--debug option)
       local_tests_only     => 0,                  # disable SA network tests (--local-only flag)
-      userstate_dir        =>
-        '/var/spool/spamassassin/spampd',         # home directory for SA files and plugins (--homedir option)
+    # Debian patch: Set userstate_dir to Debian standard. (was: '/var/spool/spamassassin/spampd')
+      userstate_dir        => '/var/cache/spampd',   # home directory for SA files and plugins (--homedir option)
+    #
       home_dir_for_helpers => '',                 # this will be set to the same as userstate_dir once options are parsed
       username             => '',                 # this will be set to the same user as we're running as once options are parsed
       userprefs_filename   => undef,              # add this config file for SA "user_prefs" settings (--saconfig option)
