--- spampd-2.52/spampd.pl.orig	2018-11-10 10:24:14.000000000 +0100
+++ spampd-2.52/spampd.pl	2018-11-12 08:32:59.000000000 +0100
@@ -145,6 +145,7 @@
     return 0 unless defined($_ = $self->_getline);
     s/[\r\n]*$//;
     $self->{state} = $_;
+    $self->{proto} = "(unknown)";
     if (s/^(l|h)?he?lo\s+//i) {  # mp: find helo|ehlo|lhlo
       # mp: determine protocol
       if (s/^helo\s+//i) {
